src/components/popover/popover.tsx

```diff
@@ -255,6 +256,10 @@ export class Popover
   //
   // --------------------------------------------------------------------------
 
+  mutationObserver: MutationObserver = createObserver("mutation", () =>
```

Comment:
As more components may need focus traps, we might need to provide a util to reuse a single mutation observer for all focus-traps.

Reply:
Agreed. Should we do this part of this PR or later?

Reply:
We can tackle this separately. https://github.com/Esri/calcite-components/issues/6668 may lead to similar components implementing `focusTrapComponent` too, so I could look into it for that one.